src/Startup.cs

```diff
@@ -32,12 +37,58 @@ public Startup(IConfiguration configuration, ILoggerFactory logFactory)
         public IConfiguration Configuration { get; }
 
         // This method gets called by the runtime. Use this method to add services to the container.
+
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddLogging(loggingBuilder => loggingBuilder.AddSerilog(dispose: true));
 
+            if (PreLanuchMode)
```

Comment:
Suggestion - can you create an extension method for IServiceCollection to do this, just to keep all the code separate? e.g. `services.ProtectOptionallyWithPassword(Configuration["SITE_PASSWORD"])`

Reply:
Type: `PreLanuchMode`

Reply:
both done, good calls